src/main/java/com/nickdsantos/onedrive4j/OneDrive.java

```diff
@@ -168,15 +168,21 @@ public AccessToken getAccessTokenFromRefreshToken(String refreshToken) throws IO
 			HttpPost httpPost = new HttpPost(uri);
 			httpPost.setEntity(formEntity);
 
-			Map<Object, Object> rawToken = httpClient.execute(httpPost, new OneDriveJsonToMapResponseHandler());
-			if (rawToken != null) {
+			Map<Object, Object> rawResponse = httpClient.execute(httpPost, new OneDriveJsonToMapResponseHandler());
+
+			if (rawResponse.containsKey("error")) {
```

Comment:
I looked at the API being called, and it seems like it can't ever be null, but they didn't document it either way, so I just moved the check.